marshal/lib/auth.py

```diff
@@ -63,10 +62,17 @@ def get_spire_svid_jwt() -> str:
     p = subprocess.run(ctx, check=True, capture_output=True)
 
     entries = json.loads(p.stdout)
+
+    file = open("/etc/cray/xname", "r")
```

Comment:
You might want a try around this call.   We could do that later.

Reply:
hmm... that's a good idea.  If for some reason the file didn't exist I wouldn't want this to abort, just have it continue and still be able to succeed in the non-xname validation case.

Reply:
Since xname validation isn't always on we could set 

```
    xname = None
    try:
        file = open("/etc/cray/xname", "r")
        xname = file.read().rstrip()
        file.close()
    except FileNotFoundError as e:
        # emit warning
        pass
```

Then in the loop
```
                if xname is not None and svid['spiffe_id'] == f"spiffe://shasta/ncn/{xname}/workload/sbps-marshal":
                    return svid['svid']
```

Reply:
This change doesn't require a full retest, just pull the little bit of code out into a test script and verify it does the right thing both when the file does and doesn't exist (don't actually remove the file, just change the file it is checking to a non-existent file)

Reply:
@don-bahls-hpe and @dborman-hpe if this is on worker node, can we get the xname from /proc/cmdline?

Reply:
FYI, I think `/etc/cray/xname` is created by node-identity script while booting and it is using the value of `xname=` from /proc/cmdline (at least compute node case).  This is why most of the Cray dracut scripts in compute node is using the value from /proc/cmdline.  I recommend using the value from /proc/cmdline to avoid any timing issues (though most likely xname file gets generated early in the boot process).